### ISSUE TITLE
Quoted properties generate invalid JavaScript

### ISSUE THREAD

**Issue by tmcw** [rollup/rollup-plugin-json#11]
I'm using rollup with the [iD](https://github.com/openstreetmap/iD) project and it's lovely! We're running into one problem importing a large JSON file that includes escaped quotes as property keys. rollup-plugin-json transforms these keys, unescaping the quotes and producing invalid JavaScript:

![2016-08-09 at 1 23 pm](https://cloud.githubusercontent.com/assets/32314/17532246/8e39dc72-5e34-11e6-8014-c716ae278749.png)

I'll push a failing testcase in a moment.

**Comment by tmcw** [comment]
Would it be better to

* Error out on this value and provide a module that exposes a JSON file only in its full form rather than exposing a tree-shaking-compatible interface?
* _Or_ doing what Elm does and automatically making the properties accessible with camelCase names?

**Comment by Victorystick** [comment]
@tmcw The first options sounds best to me. Transforming property names will only bring about more headache. How will we transform something like this in a way that is intuitive?

```json
{
  "fooBar": 1,
  "foo bar": 2,
  "foo-bar": 3
}
```

**Comment by Benjamin-Dobell** [comment]
I'm running into this in a build. It seems a `package.json` in my node_modules contains quotes inside keys/values.

However, I don't quite understand the issue, or rather the suggested resolution. Why are we talking  about either transforming keys or not converting the JSON to objects as solutions? To my knowledge JavaScript will happily accept properties with any name (including with quotes), they just can't be accessed with dot syntax, which is unimportant for tree shaking.

Isn't it just a matter of ensuring we generate valid JavaScript (i.e. using bracket syntax when keys aren't valid "identifiers")?

**Comment by tmcw** [comment]
Yes, you're misunderstanding. Modules in CommonJS are declared as properties of an object:

```js
module.exports['foo bar'] = 2;
```

But [ES6 module syntax](http://exploringjs.com/es6/ch_modules.html), module exports are identifiers, not properties:

```js
export var 'foo bar' = 2; // invalid
export var foo = 2; // valid
```

And named imports import named _identifiers_:

```js
import { 'foo bar' } from './module'; // invalid
import { foo } from './module'; // valid
```

**Comment by Benjamin-Dobell** [comment]
Ah, right, because ES6 lacks `export for as *;`

A bit obscure, but better than erroring, I imsgine you can grab all the exports that aren't valid identifiers and go:

```
var badExports = {};
badExports['not a valid identifier'] = foo;
export { badExports as __badExports };
```

Then they can still be accessed, albeit awkwardly with:

```
import { __badExports as bar };
```

**Comment by tmcw** [comment]
So! This was fixed in 0fe4fc134d3c004ca9b978cbac38ede2b780978b
